### ISSUE TITLE
The operator doesn't handle being temporarily unable to connect to the apiserver

### ISSUE THREAD

**Issue by mikebryant** [coreos/prometheus-operator#129]
If it's not able to connect, it prints this and then appears to do nothing, until the pod is deleted:
```
$ kubectl logs -f prometheus-operator-3618748121-2tgv5
ts=2017-02-01T16:32:14Z caller=main.go:121 msg="Unhandled error received. Exiting..." err="communicating with server failed: Get https://172.31.128.1:443/version: dial tcp 172.31.128.1:443: i/o timeout"
```

It should either handle this error and retry, or just exit the process so Kubernetes can restart the container etc.

**Comment by alexsomesan** [comment]
It should apparently exit, but it actually doesn't.
It looks like it may be a leaking goroutine. I'm looking into it.

**Comment by fabxc** [comment]
It doesn't exit because the other goroutines didn't terminate yet.
In particular, the HTTP server generally will never exit. I have a fixed
prepared.

**Comment by brancz** [comment]
Thanks for reporting @mikebryant this was closed by #131. It'll be included in the next release.
